core/native/refresh/src/main/csharp/Services/IconCache.cs

```diff
@@ -106,5 +121,24 @@ public bool TryGetIcon<T>(object Key, int size, out T image, string classifier =
         public ReaderWriterLockSlimExtensions.UpgradeableReadLock UpgradeableReadLock() => writer.UseUpgradeableReadLock();
 
         public ReaderWriterLockSlimExtensions.WriteLock WriteLock() => writer.UseWriteLock();
+
+        private static void OnCacheEvict(CacheEntryRemovedArguments arguments)
```

Comment:
Used to temporarily store extracted file icons (which will be garbage collected once not used in cache anymore).